openreview/venue/invitation.py

```diff
@@ -194,6 +194,145 @@ def set_submission_invitation(self):
+                            'param': {
+                                'regex': '.*', 'type': 'string' 
+                            }
+                        }
+                    }
+                },
+                'replacement': True,
+                'invitation': {
+                    'id': self.venue.get_invitation_id(invitation_name, '${2/content/noteNumber/value}'),
+                    'signatures': [venue_id],
+                    'readers': [venue_id, self.venue.get_authors_id(number='${3/content/noteNumber/value}')],
+                    'writers': [venue_id],
+                    'invitees': [venue_id, self.venue.get_authors_id(number='${3/content/noteNumber/value}')],
+                    'cdate': deletion_cdate,
+                    'process': '''def process(client, edit, invitation):
+    meta_invitation = client.get_invitation(invitation.invitations[0])
+    script = meta_invitation.content['deletion_process_script']['value']
+    funcs = {
+        'openreview': openreview,
+        'datetime': datetime
+    }
+    exec(script, funcs)
+    funcs['process'](client, edit, invitation)
+''',
+                    'edit': {
+                        'ddate': {
+                            'param': {
+                                'range': [ 0, 9999999999999 ],
+                                'optional': True                                   
+                            }
+                        },
+                        'signatures': { 
+                            'param': { 
+                                'items': [
+                                    { 'value': self.venue.get_authors_id(number='${7/content/noteNumber/value}'), 'optional': True }, 
+                                    { 'value': self.venue.get_program_chairs_id(), 'optional': True }
+                                ] 
+                            }
+                        },
+                        'readers': ['${{2/note/id}/readers}'],
+                        'writers': [venue_id, self.venue.get_authors_id(number='${4/content/noteNumber/value}')],
+                        'note': {
+                            'id': '${4/content/noteId/value}',
+                            'ddate': {
+                                'param': {
+                                    'range': [ 0, 9999999999999 ],
+                                    'optional': True,
+                                    'deletable': True
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        )
+
+        if deletion_duedate:
+            invitation.edit['invitation']['duedate'] = deletion_duedate
+
+        if deletion_expdate:
+            invitation.edit['invitation']['expdate'] = deletion_expdate
+
+        self.save_invitation(invitation, replacement=False)
+
+        expire_invitation = Invitation (
+            id=self.venue.get_invitation_id('Deletion_Expiration'),
```

Comment:
why do we need this invitation?

Reply:
ohhh I see, to expire and unexpire the existing invitations

Reply:
I'm not sure if we need to do this because the forum is marked a "deleted" so there won't be any pending tasks or buttons shown in the forum page.

Reply:
I copied this from the Withdrawal process function. We use this to expire active invitations when a paper is deleted, and then we use it to find these invitations we expired and unexpire them if the submission is restored. 
I guess we could use the Meta invitation, but I think we do this so it's easier to get these invitations if we have to unexpire them?

Reply:
I don't think we need to expire any invitation because the forum note is marked a deleted so no invitations will be valid anymore.

Reply:
Makes sense! I wasn't sure we needed it but then the tests were not passing so I thought we needed it. I'll make the change.